packages/google_maps_flutter/google_maps_flutter_platform_interface/test/types/bitmap_test.dart

```diff
@@ -237,6 +248,366 @@ void main() {
+            ]);
+          }, throwsAssertionError);
+          expect(() {
+            BitmapDescriptor.fromJson(<dynamic>[
+              AssetMapBitmap.type,
+              <String, Object?>{
+                'assetName': 'red_square.png',
+                'bitmapScaling': MapBitmapScaling.auto.name,
+                'imagePixelRatio': null,
+              }
+            ]);
+          }, throwsAssertionError);
+        });
+
+        test('with optional [width, height]', () {
+          expect(
+              BitmapDescriptor.fromJson(<dynamic>[
+                AssetMapBitmap.type,
+                <String, Object?>{
+                  'assetName': 'red_square.png',
+                  'bitmapScaling': MapBitmapScaling.auto.name,
+                  'imagePixelRatio': 1.0,
+                  'width': 1.0,
+                  'height': 1.0,
+                }
+              ]),
+              isA<BitmapDescriptor>());
+        });
+
+        test('optional width and height parameters must be in proper format',
+            () {
+          expect(() {
+            BitmapDescriptor.fromJson(<dynamic>[
+              'fromAssetImage',
+              'some/path.png',
+              'auto',
+              1.0,
+              null
+            ]);
+          }, throwsAssertionError);
+          expect(() {
+            BitmapDescriptor.fromJson(<dynamic>[
+              'fromAssetImage',
+              'some/path.png',
+              'auto',
+              1.0,
+              <dynamic>[]
+            ]);
+          }, throwsAssertionError);
+          expect(() {
+            BitmapDescriptor.fromJson(<dynamic>[
+              AssetMapBitmap.type,
+              'some/path.png',
+              <String, Object?>{
+                'assetName': 'red_square.png',
+                'bitmapScaling': MapBitmapScaling.auto.name,
+                'imagePixelRatio': null,
+                'width': null,
+                'height': 1.0,
+              }
+            ]);
+          }, throwsAssertionError);
+          expect(() {
+            BitmapDescriptor.fromJson(<dynamic>[
+              AssetMapBitmap.type,
+              'some/path.png',
+              <String, Object?>{
+                'assetName': 'red_square.png',
+                'bitmapScaling': MapBitmapScaling.auto.name,
+                'imagePixelRatio': null,
+                'width': 1.0,
+                'height': null,
+              }
+            ]);
+          }, throwsAssertionError);
+          expect(() {
+            BitmapDescriptor.fromJson(<dynamic>[
+              AssetMapBitmap.type,
+              'some/path.png',
+              <String, Object?>{
+                'assetName': 'red_square.png',
+                'bitmapScaling': MapBitmapScaling.auto.name,
+                'imagePixelRatio': null,
+                'width': '1.0',
+              }
+            ]);
+          }, throwsAssertionError);
+        });
+      });
+    });
+  });
+
+  group('$AssetMapBitmap', () {
```

Comment:
We're moving away from using `$` in test and group names, since it confuses the VS Code test explorer, so new code should just use regular strings.